src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropdownMenu';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setSelectPerson(value);
+
+    if (value === '') {
+      setAppliedQuery('');
+    } else {
+      applyQuery(value);
+    }
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name
+        .toLowerCase()
+        .trim()
+        .includes(appliedQuery.toLowerCase().trim()),
+    );
+  }, [appliedQuery]);
+
+  const { name, born, died } = filteredPeople[0] || [];
```

Comment:
In this line, you directly interact with the DOM by getting a reference to the first person from the filtered list. It is better to use React to manage your state and data flow. Instead, you can pass the selected person as a prop to the child components.